plugin/symbols.py

```diff
@@ -156,3 +159,61 @@ def process_symbol_informations(self, items: List[Dict[str, Any]]) -> List[List[
             self.regions.append((range_to_region(Range.from_lsp(item['location']['range']), self.view), None))
             quick_panel_items.append([item['name'], format_symbol_kind(item['kind'])])
         return quick_panel_items
+
+
+class SymbolQueryInput(sublime_plugin.TextInputHandler):
+
+    def validate(self, txt: str) -> bool:
+        return txt != ""
+
+    def placeholder(self) -> str:
+        return "Symbol"
+
+
+class LspWorkspaceSymbolsCommand(LspTextCommand):
+    def __init__(self, view: sublime.View) -> None:
+        super().__init__(view)
+
+    def _format(self, s: Dict[str, Any]) -> str:
+        file_name = os.path.basename(s['location']['uri'])
+        symbol_kind = format_symbol_kind(s["kind"])
+        name = "{} ({}) - {} -- {}".format(s['name'], symbol_kind, s.get('containerName', ""), file_name)
+        return name
+
+    def _open_file(self, symbols: List[Dict[str, Any]], index: int) -> None:
+        if index != -1:
+            symbol = symbols[index]
+            window = self.view.window()
+            if window:
+                window.open_file(location_to_encoded_filename(symbol['location']), sublime.ENCODED_POSITION)
+
+    def _handle_response(self, query: str, response: Optional[List[Dict[str, Any]]]) -> None:
```

Comment:
Me being boring again: would love those to be ordered better (moved after `run`).